_includes/contact_form.html

```diff
@@ -399,69 +406,21 @@
+          {{ common_application | translate: 'settings.contact_page.support_request_form.agencies' }}
+        </option>
+      {% endfor %}
     </optgroup>
     <optgroup
       label="{{ site.data.[page.lang].settings.contact_page.support_request_form.labels.agency_application_all }}"
     >
-      <option value="Customs and Border Protection Jobs">
-        {{ site.data.[page.lang].settings.contact_page.support_request_form.agencies.cbp_jobs }}
-      </option>
-      <option value="Customs and Border Protection ROAM">
-        {{ site.data.[page.lang].settings.contact_page.support_request_form.agencies.cbp_roam }}
-      </option>
-      <option value="DOD Moves.mil (Homes or Office)">
-        {{ site.data.[page.lang].settings.contact_page.support_request_form.agencies.move_mil }}
-      </option>
-      <option value="DOT FMCSA (Motor Carrier Safety)">
-        {{ site.data.[page.lang].settings.contact_page.support_request_form.agencies.dot_fmcsa }}
-      </option>
-      <option value="DOT National Registry of Certified Medical Examiners">
-        {{ site.data.[page.lang].settings.contact_page.support_request_form.agencies.dot_nrcme }}
-      </option>
-      <option value="DS Logon">
-        {{ site.data.[page.lang].settings.contact_page.support_request_form.agencies.ds_logon }}
-      </option>
-      <option value="GSA Sam.gov">
-        {{ site.data.[page.lang].settings.contact_page.support_request_form.agencies.sam }}
-      </option>
-      <option value="Railroad Retirement Board">
-        {{
-        site.data.[page.lang].settings.contact_page.support_request_form.agencies.railroad_retirement_board
-        }}
-      </option>
-      <option value="SBA DLAP">
-        {{ site.data.[page.lang].settings.contact_page.support_request_form.agencies.sba_dlap }}
-      </option>
-      <option value="Social Security Administration (SSA)">
-        {{ site.data.[page.lang].settings.contact_page.support_request_form.agencies.ssa }}
-      </option>
-      <option value="Trusted Traveler Programs (Global Entry/Nexus/Sentri)">
-        {{
-        site.data.[page.lang].settings.contact_page.support_request_form.agencies.trusted_traveler_program
-        }}
-      </option>
-      <option value="USAJOBS">
-        {{ site.data.[page.lang].settings.contact_page.support_request_form.agencies.usajobs }}
-      </option>
-      <option value="Veterans Affairs">
-        {{
-        site.data.[page.lang].settings.contact_page.support_request_form.agencies.veterans_affairs
-        }}
-      </option>
+      {% assign all_applications_sorted = include.all_applications | sort_translated: 'settings.contact_page.support_request_form.agencies' %}
+      {% for application in all_applications_sorted %}
+        <option value="{{ application }}">
+          {{ application | translate: 'settings.contact_page.support_request_form.agencies' }}
+        </option>
+      {% endfor %}
```

Comment:
If we do the key/value idea, to translate by sorted labels, we just need to sort by the values of the hash. The syntax wasn't very obvious, but it looks like this works?

```suggestion
      {% assign all_applications_sorted = (site.data[page.lang].settings.contact_page.support_request_form.agencies | sort: 1) %}
      {% for application in all_applications_sorted %}
        <option value="{{ application[0] }}">
          {{ application[1] }}
        </option>
      {% endfor %}
```

Reply:
will try it out!

Reply:
I had some problems with this? It didn't like the `sort: 1` as far as I could tell, so I ended up adding a `sort_by_values` filter in 8e178e635e80f4eaa37ea918a430b2abdda5f670 which let me return objects with `.key` and `.value` which made it slightly easier to read

Reply:
Hm, yeah, I tested this a bit more and it's sort of odd. Spelunking in the Jekyll source, I guess we need to pass both parameters for this for the numeric sort key to be picked up, like `| sort: 1, 'first'`, but even then it's [coerced back to a string](https://github.com/jekyll/jekyll/blob/76517175e700d80706c9139989053f1c53d9b956/lib/jekyll/filters.rb#L425), so it results in an error "no implicit conversion of String into Integer" (i.e. `['key', 'value']['1']`).

Kinda a pain we have to make our own filter, but I agree it makes it more readable.